Actions/TelemetryHelper.psm1

```diff
@@ -212,4 +239,41 @@ function Add-TelemetryProperty() {
     }
 }
 
-Export-ModuleMember -Function Trace-Information, Trace-Exception, Add-TelemetryProperty
+<#
+    .SYNOPSIS
+    Writes a deprecation warning to telemetry and as a warning in the GitHub action
+
+    .DESCRIPTION
+    Writes a deprecation warning to telemetry and as a warning in the GitHub action
+
+    .PARAMETER Message
+    The message to log to telemetry
+
+    .PARAMETER DeprecationTag
+    The tag to use to link to the deprecation documentation
+
+    .PARAMETER AdditionalData
+    Additional data to log to telemetry
+
+    .EXAMPLE
+    Trace-DeprecationWarning -Message "This setting is deprecated. Use the new setting instead." -DeprecationTag 'SettingName'
+#>
+function Trace-DeprecationWarning {
+    param(
+        [Parameter(Mandatory = $true)]
+        [String] $Message,
+        [Parameter(Mandatory = $true)]
+        [String] $DeprecationTag,
+        [Parameter(Mandatory = $false)]
+        [System.Collections.Generic.Dictionary[[System.String], [System.String]]] $AdditionalData = @{}
```

Comment:
Why is this not a hashtable type? (system.collections.gene...)

Reply:
Not sure I remember why but I can check up on that. I can create a follow-up PR for that. 

Right now, this type is consistent with what we use for tracing "regular" telemetry as well so keeping it in for now.